protocol/messenger.go

```diff
@@ -2940,6 +2940,30 @@ func (r *ReceivedMessageState) addNewMessageNotification(publicKey ecdsa.PublicK
 	return nil
 }
 
+// updateActivityCenterNotification updates AC notification if it exists and hasn't been read yet
+func (r *ReceivedMessageState) updateActivityCenterNotification(publicKey ecdsa.PublicKey, m *Messenger, message *common.Message, responseTo *common.Message) error {
```

Comment:
I have a tendency to prefer to separate the CRUD operations into separate functions if possible, so that the functions are more flexible and can end up being used in different contexts.

In this concrete case, the function says it updates the entity, but it also reads it first from the db. Instead, we could separate the read from the update, i.e. the function `updateActivityCenterNotification` would receive an instance of `ActivityCenterNotification`.

There's only a marginal gain for this PR, so take my suggestion with a grain of salt, but applying this practice throughout the system could be quite beneficial.

Reply:
now it says that it updates existing entry, I consider the case closed :D